File: Figure/src/figure/rectangle.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package figure;

/**
 *
 * @author eyro1
 */
public class rectangle implements figure {
    private String color;
    private double length, width;

    public rectangle(String color, double length, double width) {
        this.color = color;
        this.length = length;
        this.width = width;
    }

    @Override
    public String getColor() {
        return color;
    }

    @Override
    public String getName() {
        return "Rectangle";
    }

    @Override
    public double perimeter() {
        return 2 * (length + width);
    }

    @Override
    public double area() {
        return length * width;
    }
}

File: rectangle.py
from figure import Figure

class Rectangle(Figure):
    def __init__(self, color: str, length: float, width: float):
        self.color = color
        self.length = length
        self.width = width

    def getColor(self):
        return self.color

    def get_name(self):
        return "Rectangle"

    def perimeter(self):
        return 2 * (self.length + self.width)

    def area(self):
        return self.length * self.width
